cmd/auth-rpc-client.go

```diff
@@ -146,21 +142,34 @@ func (authClient *AuthRPCClient) Call(serviceMethod string, args interface {
 	SetToken(token string)
 	SetTimestamp(tstamp time.Time)
 }, reply interface{}) (err error) {
-	// On successful login, attempt the call.
-	if err = authClient.Login(); err == nil {
-		// Set token and timestamp before the rpc call.
-		args.SetToken(authClient.serverToken)
-		args.SetTimestamp(time.Now().UTC())
-
-		// Call the underlying rpc.
-		err = authClient.rpc.Call(serviceMethod, args, reply)
-
-		// Invalidate token, and mark it for re-login on subsequent reconnect.
-		if err == rpc.ErrShutdown {
-			authClient.mu.Lock()
-			authClient.isLoggedIn = false
-			authClient.mu.Unlock()
+	loginAndCallFn := func() error {
+		// On successful login, proceed to attempt the requested service method.
+		if err = authClient.Login(); err == nil {
+			// Set token and timestamp before the rpc call.
+			args.SetToken(authClient.serverToken)
+			args.SetTimestamp(time.Now().UTC())
+
+			// Finally make the network call using net/rpc client.
+			err = authClient.rpc.Call(serviceMethod, args, reply)
 		}
+		return err
+	}
```

Comment:
The practice is not to any specific language and Golang doesn't have such enforcement about not to have block separation.

I think you refer Golang as its library code base.  I see golang internal libraries do not follow such convention because of very specific/simpler/few line of code within a function, but golang tools follow that style.

I checked Kubernetes/Docker applications written in Go.  They follow the style I mentioned.

ATM there is no tool can beautify a code by logic, such tool is hard to write :smile: